src/connector/src/sink/mongodb.rs

```diff
@@ -0,0 +1,802 @@
+            // there are some programming error in our code.
+            client_copy.shutdown().await;
+            tracing::debug!(%name, "client shutdown succeeded");
+        });
+        Self { tx, client }
+    }
+}
+
+impl Deref for ClientGuard {
+    type Target = Client;
+
+    fn deref(&self) -> &Self::Target {
+        &self.client
+    }
+}
+
+#[derive(Debug)]
+pub struct MongodbSink {
+    pub config: MongodbConfig,
+    param: SinkParam,
+    schema: Schema,
+    pk_indices: Vec<usize>,
+    is_append_only: bool,
+}
+
+impl MongodbSink {
+    pub fn new(param: SinkParam) -> Result<Self> {
+        let config = MongodbConfig::from_hashmap(param.properties.clone())?;
+        let pk_indices = param.downstream_pk.clone();
+        let is_append_only = param.sink_type.is_append_only();
+        let schema = param.schema();
+        Ok(Self {
+            config,
+            param,
+            schema,
+            pk_indices,
+            is_append_only,
+        })
+    }
+}
+
+impl TryFrom<SinkParam> for MongodbSink {
+    type Error = SinkError;
+
+    fn try_from(param: SinkParam) -> std::result::Result<Self, Self::Error> {
+        MongodbSink::new(param)
+    }
+}
+
+impl Sink for MongodbSink {
+    type Coordinator = DummySinkCommitCoordinator;
+    type LogSinker = LogSinkerOf<MongodbSinkWriter>;
+
+    const SINK_NAME: &'static str = MONGODB_SINK;
+
+    fn is_sink_decouple(desc: &SinkDesc, user_specified: &SinkDecouple) -> Result<bool> {
+        match user_specified {
+            SinkDecouple::Default => Ok(desc.sink_type.is_append_only()),
+            SinkDecouple::Disable => Ok(false),
+            SinkDecouple::Enable => Ok(true),
+        }
+    }
+
+    async fn validate(&self) -> Result<()> {
+        if !self.is_append_only {
+            if self.pk_indices.is_empty() {
+                return Err(SinkError::Config(anyhow!(
+                "Primary key not defined for upsert mongodb sink (please define in `primary_key` field)")));
+            }
+            if self.pk_indices.len() != 1 {
+                return Err(SinkError::Config(anyhow!(
+                    "upsert mongodb sink requires exactly one primary field as the _id field"
+                )));
+            }
+        }
+
+        if self.config.bulk_write_max_entries > MONGODB_BULK_WRITE_SIZE_LIMIT {
+            return Err(SinkError::Config(anyhow!(
+                "mongodb.bulk_write.max_entries {} exceeds the limit {}",
+                self.config.bulk_write_max_entries,
+                MONGODB_BULK_WRITE_SIZE_LIMIT
+            )));
+        }
+
+        if let Err(err) = self.config.common.collection_name.parse::<Namespace>() {
+            return Err(SinkError::Config(anyhow!(err).context(format!(
+                "invalid collection.name {}",
+                self.config.common.collection_name
+            ))));
+        }
+
+        // checking reachability
+        let client = self.config.common.build_client().await?;
```

Comment:
Is there any other issue here, such as the data already existing or overwriting the user's data? If so, validation can be omitted

Reply:
Could you elaborate more? I’m not quite getting your point

Reply:
I am concerned that sending non user data may cause issues such as data coverage

Reply:
do you mean sending the "hello" command?

Reply:
This command should be safe to test the connectivity, it will not modify the user's data.